Magic buttons for the joyop

stonier opened the pull request:
These connect the button to a `std_mgs::Bool` publisher.

stonier committed:
joyop magic buttons

stonier committed:
[yocs_joyop] launcher args for the magic topic names.

bit-pirate commented on yocs_joyop/launch/joyop.launch:
```diff
@@ -5,33 +5,41 @@
-	<arg name="cmd_vel_topic_name"  default="cmd_vel"/>
-	<arg name="enable_topic_name"   default="enable"/>
-	<arg name="disable_topic_name"  default="disable"/>
+    <arg name="linear_axis"           default="1"/>
```
Formatting error? I'm using the roscpp styleguide for the launchers as well, since I don't know any guide better suited.

stonier replied:
There were tabs in it before, so I just let eclipse roll over it with its default formatting for xml files which is four spaces. Didn't know there was a ros convention, let's use that.

bit-pirate commented on yocs_joyop/launch/joyop.launch:
```diff
@@ -5,33 +5,41 @@
+		<param name="deadman_button"        value="$(arg deadman_button)"/>
+		<param name="enable_button"         value="$(arg enable_button)"/>
+		<param name="disable_button"        value="$(arg disable_button)"/>
+		<param name="magic_button"          value="$(arg magic_button"/>
```
This should throw an error. Missing a ")" at the end.

stonier replied:
Odd, should have thrown an error for me.

bit-pirate commented on yocs_joyop/launch/joyop.launch:
```diff
@@ -5,33 +5,41 @@
+		<param name="enable_button"         value="$(arg enable_button)"/>
+		<param name="disable_button"        value="$(arg disable_button)"/>
+		<param name="magic_button"          value="$(arg magic_button"/>
+		<param name="more_magic_button"     value="$(arg more_magic_button"/>
```
Same here.

bit-pirate commented on yocs_joyop/src/joyop.cpp:
```diff
@@ -136,12 +157,28 @@ void JoyOp::joyCallback(const sensor_msgs::Joy::ConstPtr& joy)
 {
   boost::mutex::scoped_lock lock(publish_mutex_);
 
+  if ( magic_ != magic_pressed_ ) {
```
Shouldn't `magic_pressed_` and `more_magic_pressed_` be initialised in the constructor as well?

stonier replied:
Yeah, missed them because they were set elsewhere. Cleaning them up now.

stonier committed:
[yocs_joyop] magic buttons now timestamped

Uses the magic button message type, commited in
https://github.com/yujinrobot/yocs_msgs/commit/271650749afc4c19c158d4e40be923cb670e1a26

stonier committed:
[yocs_joyop] clean up xml formatting, fix typo and innitialise variables.

stonier commented:
Cleaned up, and added timestamps to the events. Uses the `MagicButton.msg` commited at https://github.com/yujinrobot/yocs_msgs/commit/271650749afc4c19c158d4e40be923cb670e1a26.

bit-pirate commented:
It's a pity that we need a custom message for this, but I don't see a better way either.
